geodata/models/fmv/base.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.gis.db import models
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from s3_file_field import S3FileField
+
+from ... import tasks
+from ..common import ChecksumFile, ModifiableEntry, SpatialEntry
+from ..constants import DB_SRID
+from ..mixins import TaskEventMixin
+
+
+class FMVFile(ChecksumFile, TaskEventMixin):
+    """For uploading single FMV files (mp4)."""
+
+    task_func = tasks.task_read_fmv_file
+    failure_reason = models.TextField(null=True, blank=True)
+    file = S3FileField(upload_to='files/fmv/')
+
+
+@receiver(post_save, sender=FMVFile)
+def _post_save_fmv_file(sender, instance, *args, **kwargs):
+    transaction.on_commit(lambda: instance._post_save_event_task(*args, **kwargs))
+
+
+class FMVEntry(ModifiableEntry, SpatialEntry):
+    """Single FMV entry, tracks the original file."""
+
+    def __str__(self):
+        return f'{self.name} ({self.id})'
+
+    name = models.CharField(max_length=100, blank=True, null=True)
+    description = models.TextField(null=True, blank=True)
+
+    fmv_file = models.OneToOneField(FMVFile, null=True, on_delete=models.CASCADE)
+    klv_file = S3FileField(upload_to='files/fmv/', null=True, blank=True)
+    web_video_file = S3FileField(upload_to='files/fmv/web/', null=True, blank=True)
```

Comment:
@manthey, how do you feel about this? We cannot display the raw videos in a web browser and these videos are significant in size, so I added a routine to use `ffmpeg` to convert the videos to a web-friendly format on import and save that video here (this is effectively duplicate data through all the spatial metadata in the video is lost during this conversion)

Reply:
I think it is the right thing to do.  I'm wondering if we should optimize the video for web playback as well as converting its format.  That is, for easy seeking, it is best to have a gop so that there is a keyframe every half second or so.  Similarly, we may want to have a maximum bitrate and other parameters.  And, using the copy codec is fast, but not guaranteed that it will play on a web browser. 

Other options might also be appropriate (e.g., `ffmpeg -i <source> -map_metadata -1 -vcodec libx264 -level 30 -maxrate 1024k -movflags faststart -g 15 -an <output>`)

Reply:
I updated the conversion to use those options and it drastically reduced the video file size

Reply:
The one that most affects size is `-maxrate`.  The value I picked will work on even very slow connections, but I could see wanting higher quality at the cost of not working on slow connections.  DVDs, for example, have a typical average bitrate of 6144k (if I remember correctly, excluding audio) and a typical maximum rate of 10000k.  Here we specify the maximum rate; you can also specify more parameters to try to target an average rate via `-vb`.  So typical DVD quality would be `-vb 6144k -maxrate 10000k`.  And, for what it is worth, smaller gop (`-g`) will result in larger files that seek better.

Reply:
@manthey, is this changing the rate and thus the number of frames in the output video? We may want to keep the same number of frames so that we can correspond those frames to the stored footprint polygons more easily for #146. Another approach would be to store the timestamp of the video with the footprint polygons to avoid worrying about the number of frames and to make it easier to implement #146 based on time rather than frame

Reply:
No, this isn't changing the number of frames.  It is recompressing the video and changing which frames are key frames.

There are videos that have non-constant frame rates.  If we ever need to know the precise time of each frame, we'd have to store the frame time with each footprint.  In practice, most videos from cameras are constant frame rate (as opposed to videos made of power-point presentations).